starport_template/lib/widgets/transaction_history_list.dart

```diff
@@ -0,0 +1,55 @@
+import 'package:cosmos_ui_components/components/template/transaction_history_card.dart';
+import 'package:cosmos_ui_components/components/template/transaction_history_group_title.dart';
+import 'package:cosmos_ui_components/utils/date_formatter.dart';
+import 'package:cosmos_utils/cosmos_utils.dart';
+import 'package:flutter/material.dart';
+import 'package:starport_template/entities/transaction_history_item.dart';
+
+class TransactionHistoryList extends StatefulWidget {
+  final List<TransactionHistoryItem> transactionsList;
+
+  const TransactionHistoryList({Key? key, required this.transactionsList}) : super(key: key);
+
+  @override
+  _TransactionHistoryListState createState() => _TransactionHistoryListState();
+}
+
+class _TransactionHistoryListState extends State<TransactionHistoryList> {
+  late Map<String, List<TransactionHistoryItem>> groupedHistory = {};
+
+  @override
+  void initState() {
+    super.initState();
+    groupedHistory.addAll(widget.transactionsList.groupBy((m) => formatDate(m.date, DateFormatEnum.groupByMonthYear)));
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return CustomScrollView(
+      slivers: groupedHistory.keys.map(
+        (groupedMonthYear) {
+          final transactionsList = groupedHistory[groupedMonthYear] ?? [];
+          return SliverToBoxAdapter(
+            child: Column(
+              crossAxisAlignment: CrossAxisAlignment.start,
+              children: [
+                TransactionHistoryGroupTitle(title: groupedMonthYear.split('-')[0]),
```

Comment:
like I've mentioned, we don't want to have any string splitting logic inside the view, so lets at least extract it to separate method, which name describes why this splitting is needed. Ideally you could create the method adjacent to the `formatDate` and give it descriptive name